### PR TITLE
M3-1433: add tags input to Domain drawer

### PR DESCRIPTION
**Pull request by rulikrulit**
- Moved tags input to a separate component
- Simplified usage and logic, removed TagsHOC component
- Added tags input to Domains drawer form

### CODE REVIEW

File: src/components/TagsInput/TagsInput.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    onChange(updatedSelectedTags);
+  }
+
+  tagToItem = (tag:string) => {
+    return { value: tag, label: tag }
+  }
+
+  state = {
+    accountTags: [],
+    selectedTags: [],
```
Comment (Jskobos) [inline src/components/TagsInput/TagsInput.tsx]: This is the only reference to this value. Is it going to be used anywhere in the future?

File: src/components/TagsInput/TagsInput.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      accountTags: updatedTags,
+    });
+    onChange(updatedSelectedTags);
+  }
+
+  tagToItem = (tag:string) => {
+    return { value: tag, label: tag }
+  }
+
+  state = {
```
Comment (Jskobos) [inline src/components/TagsInput/TagsInput.tsx]: Should be `state: State = {...}` (see above)

File: src/components/TagsInput/TagsInput.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  label: string;
+}
+
+export interface Props {
+  tagError?: string;
+  value: Item[];
+  onChange: (selected: Item | Item[]) => void;
+}
+
+class TagsInput extends React.Component<Props> {
```
Comment (Jskobos) [inline src/components/TagsInput/TagsInput.tsx]: Please declare a State interface and use it here (e.g. `React.Component<Props, State>`)

File: src/components/TagsInput/TagsInput.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+export interface Tag {
+  value: string;
+  label: string;
+}
+
+export interface Props {
+  tagError?: string;
+  value: Item[];
+  onChange: (selected: Item | Item[]) => void;
```
Comment (Jskobos) [inline src/components/TagsInput/TagsInput.tsx]: If this is only ever a MultiSelect, onChange can probably be `(selected: Item[]) => void`. Not 100% sure...

Comment (rulikrulit) [reply src/components/TagsInput/TagsInput.tsx]: I think it should always be multi so you are right, fixing this one too

File: src/components/TagsInput/TagsInput.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+
+export interface Tag {
+  value: string;
+  label: string;
+}
+
+export interface State {
+  accountTags: Item[],
+  errors: Linode.ApiFieldError[] | undefined,
```
Comment (Jskobos) [inline src/components/TagsInput/TagsInput.tsx]: Could be more consistent about what the empty error state is. You initialize to `[]` (which is fine) but then reset it to `undefined` (which is also fine, but different).

Comment (rulikrulit) [reply src/components/TagsInput/TagsInput.tsx]: Agree, empty array is good for consistency, going to update it

**Pull request merged**